tinysegmenter/tinysegmenter.py

```diff
@@ -12,181 +12,227 @@
+_UW5 = {",":465,".":-299,"1":-514,"E2":-32768,"]":-2762,"、":465,"。":-299,"「":363,"あ":1655,"い":331,"う":-503,"え":1199,"お":527,"か":647,"が":-421,"き":1624,"ぎ":1971,"く":312,"げ":-983,"さ":-1537,"し":-1371,"す":-852,"だ":-1186,"ち":1093,"っ":52,"つ":921,"て":-18,"で":-850,"と":-127,"ど":1682,"な":-787,"に":-1224,"の":-635,"は":-578,"べ":1001,"み":502,"め":865,"ゃ":3350,"ょ":854,"り":-208,"る":429,"れ":504,"わ":419,"を":-1264,"ん":327,"イ":241,"ル":451,"ン":-343,"中":-871,"京":722,"会":-1153,"党":-654,"務":3519,"区":-901,"告":848,"員":2104,"大":-1296,"学":-548,"定":1785,"嵐":-1304,"市":-2991,"席":921,"年":1763,"思":872,"所":-814,"挙":1618,"新":-1682,"日":218,"月":-4353,"査":932,"格":1356,"機":-1508,"氏":-1347,"田":240,"町":-3912,"的":-3149,"相":1319,"省":-1052,"県":-4003,"研":-997,"社":-278,"空":-813,"統":1955,"者":-2233,"表":663,"語":-1073,"議":1219,"選":-1018,"郎":-368,"長":786,"間":1191,"題":2368,"館":-689,"１":-514,"Ｅ２":-32768,"｢":363,"ｲ":241,"ﾙ":451,"ﾝ":-343};
+_UW6 = {",":227,".":808,"1":-270,"E1":306,"、":227,"。":808,"あ":-307,"う":189,"か":241,"が":-73,"く":-121,"こ":-200,"じ":1782,"す":383,"た":-428,"っ":573,"て":-1014,"で":101,"と":-105,"な":-253,"に":-149,"の":-417,"は":-236,"も":-206,"り":187,"る":-135,"を":195,"ル":-673,"ン":-496,"一":-277,"中":201,"件":-800,"会":624,"前":302,"区":1792,"員":-1212,"委":798,"学":-960,"市":887,"広":-695,"後":535,"業":-697,"相":753,"社":-507,"福":974,"空":-822,"者":1811,"連":463,"郎":1082,"１":-270,"Ｅ１":306,"ﾙ":-673,"ﾝ":-496};
+
+def _ctype(c):
+    return _CHARDICT.get(c, 'O')
+
+def tokenize(text):
+    if text == "":
+        return []
+
+    result = []
+    seg = ["B3","B2","B1"]
+    ctype = ["O","O","O"]
+    seg.extend(text)
+    ctype.extend(map(_ctype, text))
```

Comment:
list comprehension is much faster in this example below.
Could you consider using list comprehension ?

```py
In [52]: %timeit l.extend(map(lambda x:x, text))
The slowest run took 8.03 times longer than the fastest. This could mean that an intermediate result is being cached
1000000 loops, best of 3: 771 ns per loop

In [56]: %timeit [ l.append( (lambda t:t)(t) ) for t in text]
The slowest run took 4.17 times longer than the fastest. This could mean that an intermediate result is being cached
1000000 loops, best of 3: 1.27 µs per loop
```

~~~py
In [62]: l = []

In [63]:  l.extend(map(lambda x:x, text))

In [64]: l
Out[64]: ['h', 'o', 'g', 'e']

In [65]: l = []

In [66]: [ l.append( (lambda t:t)(t) ) for t in text]
Out[66]: [None, None, None, None]

In [67]: l
Out[67]: ['h', 'o', 'g', 'e']
~~~